oauthproxy.go

```diff
@@ -885,6 +887,12 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 		} else {
 			req.Header.Del("X-Forwarded-Preferred-Username")
 		}
+
+		if len(session.Groups) > 0 {
+			req.Header["X-Forwarded-Groups"] = []string{p.encodeArrayHeader(session.Groups)}
+		} else {
+			req.Header.Del("X-Forwarded-Groups")
```

Comment:
You'll need to handle Group header management in this new secure function as well: https://github.com/oauth2-proxy/oauth2-proxy/blob/2318716a8992686a31f4fadaa1ec3b12ff2dd22b/oauthproxy.go#L955

Reply:
Should be done.